test/lib/tokenFactory.test.js

```
@@ -0,0 +1,187 @@
+            lastUsed: null
+        }, {
+            id: 456,
+            userId: 4321,
+            name: 'token2',
+            description: 'token number 2',
+            value: 'sealedToken2',
+            lastUsed: '2017-05-11T22:48:16.827Z'
+        }];
+
+        const returnValue = [{
+            id: 123,
+            userId: 4321,
+            name: 'token1',
+            description: 'token number 1',
+            value: 'unsealedToken1',
+            lastUsed: null
+        }, {
+            id: 456,
+            userId: 4321,
+            name: 'token2',
+            description: 'token number 2',
+            value: 'unsealedToken2',
+            lastUsed: '2017-05-11T22:48:16.827Z'
+        }];
+
+        it('calls datastore scan and returns correct values', () => {
+            datastore.scan.resolves(datastoreReturnValue);
+            ironMock.unseal.withArgs('sealedToken1', password).yieldsAsync(null, 'unsealedToken1');
+            ironMock.unseal.withArgs('sealedToken2', password).yieldsAsync(null, 'unsealedToken2');
+
+            return factory.list({ paginate, params })
+                .then((arr) => {
+                    assert.isArray(arr);
+                    assert.equal(arr.length, 2);
+                    assert.deepEqual(arr, returnValue);
+                    arr.forEach((model) => {
+                        assert.instanceOf(model, Token);
+                    });
+                });
+        });
+    });
+
+    describe('getInstance', () => {
+        let config;
+
+        beforeEach(() => {
+            config = { datastore };
+        });
+
+        it('should get an instance', () => {
+            const f1 = TokenFactory.getInstance(config);
+            const f2 = TokenFactory.getInstance(config);
+
+            assert.instanceOf(f1, TokenFactory);
+            assert.instanceOf(f2, TokenFactory);
+
+            assert.equal(f1, f2);
+        });
+
+        it('should throw when config not supplied', () => {
```

"should throw an error when ..."